### ISSUE TITLE
Subtitles display problem when not PGS

### ISSUE THREAD

**Issue by bouchnouk** [justdan96/tsMuxer#190]
Hi,

I tried to remux several MKV files with subtitles and I am getting nothing outputted whenever SRT type is in the file. When a subtitle with PGS format is in the file, the subtitles are displayed correctly when playing the video result. I am using default values in Subtitles tab. I am also using Open Sans font.

I am sending metafiles of MKV files with SRT subtitles giving incorrect output.

Is this a bug?

[Dolemite.Is.My.Name.2019.1080p.DDP5.1.x264.meta.zip](https://github.com/justdan96/tsMuxer/files/4183916/Dolemite.Is.My.Name.2019.1080p.DDP5.1.x264.meta.zip)
[The.Irishman.2019.1080p.DDP5.1.Atmos.H.265.HDR.meta.zip](https://github.com/justdan96/tsMuxer/files/4183920/The.Irishman.2019.1080p.DDP5.1.Atmos.H.265.HDR.meta.zip)

**Comment by bouchnouk** [comment]
I did a lit bit more investigation and I found that the subtitles causing problems are in UTF-8 format. PGS format doesn't cause any trouble when playing the video on Bluray player or Mac.
 
<img width="700" alt="2020-02-10_22-18-47" src="https://user-images.githubusercontent.com/20449937/74208868-50cf9780-4c53-11ea-8bd2-5224fdce7377.png">

**Comment by justdan96** [comment]
#186 gave us support for UTF8 SRT files. Is there anything that is output to the console when you try to mux the file?

**Comment by bouchnouk** [comment]
This is what I get when remuxing a file with UTF8 subtitles:

<img width="633" alt="2020-02-11_19-58-50" src="https://user-images.githubusercontent.com/20449937/74293469-ef6a0000-4d08-11ea-9710-8e157e2f5fc6.png">

**Comment by bouchnouk** [comment]
This is the information of the MKV file with UTF8 subtitles. I have noticed that tsMuxer seems to see UTF8 subtitles as PGS. That's why we see "garbage" on the screen because the real content of the subtitles file is displayed (binary) instead of the subtitles text. 

<img width="700" alt="2020-02-11_20-00-44" src="https://user-images.githubusercontent.com/20449937/74293551-335d0500-4d09-11ea-9223-34d7287eb552.png">

**Comment by justdan96** [comment]
@xavery any idea what's going on there?

**Comment by xavery** [comment]
> @xavery any idea what's going on there?

The metafiles look okay to me, so I suspect this is somehow related to how the SRT tracks from the Matroska container are re-muxed into the transport stream. Perhaps some alternative code path is taken in this case, and I wasn't aware of this. I'll have to try and reproduce this myself first in order to say for sure.

**Comment by xavery** [comment]
This isn't related to UTF-8 or Matroska. I reproduced the issue with the MP4 version of BBB and [this subtitle file](https://www.opensubtitles.org/en/subtitles/8102110/mixed-ish-she-works-hard-for-the-money-en), using the following metafile :
```
MUXOPT --no-pcr-on-video-pid --new-audio-pes --vbr  --vbv-len=500
V_MPEG4/ISO/AVC, "/home/daniel/bbb_sunflower_1080p_30fps_normal.mp4", insertSEI, contSPS, track=1, lang=und
A_MP3, "/home/daniel/bbb_sunflower_1080p_30fps_normal.mp4", track=2, lang=und
A_AC3, "/home/daniel/bbb_sunflower_1080p_30fps_normal.mp4", track=3, lang=und
S_TEXT/UTF8, "/home/daniel/Mixed-ish.S01E16.720p.HDTV.x264-AVS+SVA.HI.C.srt",font-name="Arial",font-size=65,font-color=0xffffffff,bottom-offset=24,font-border=5,text-align=center,video-width=1920,video-height=1080,fps=25
```
[This](https://files.catbox.moe/qmrh0c.png) is the effect. The garbled subtitles can be seen in the lower bottom corner.

Bisecting leads to e58b4de12c86bf5b8d6f35b79337b86d2e4574fe , which sucks, because it fixed another issue. @jcdr428 can you have a look?

**Comment by jcdr428** [comment]
@xavery should be solved with #198 .

Before e58b4de:
<img width="196" alt="a4f5a77" src="https://user-images.githubusercontent.com/56721609/75018098-004c0b80-5497-11ea-824f-9c12da2b98be.png">

After this commit:
<img width="183" alt="after commit" src="https://user-images.githubusercontent.com/56721609/75018221-283b6f00-5497-11ea-8f10-942f37fd8e38.png">
